### PR TITLE
[Custom loops PoC] Extend compatibility to tests

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
# Description

- Extends compatibility to tests.

### CODE REVIEW

File: src/anomalib/trainer/connectors/thresholding_connector.py [hunk 1]
```diff
@@ -72,12 +71,13 @@ def compute(self):
...
-            self.trainer.pixel_threshold.compute()
-        else:
-            self.trainer.pixel_threshold.value = self.trainer.image_threshold.value
+        if self.threshold_method == ThresholdMethod.ADAPTIVE:
+            if self.trainer.image_threshold is not None:
+                self.trainer.image_threshold.compute()
+            if self.trainer.task_type in (TaskType.SEGMENTATION, TaskType.DETECTION):
+                self.trainer.pixel_threshold.compute()
+            else:
+                self.trainer.pixel_threshold.value = self.trainer.image_threshold.value
```
Comment (samet-akcay) [inline src/anomalib/trainer/connectors/thresholding_connector.py:81]: This part will change with the thresholding refactor, right?

**Review by samet-akcay — COMMENTED** [review]
Thanks, just two comments.

File: src/anomalib/trainer/connectors/thresholding_connector.py (hunk 1 above)
Comment (ashwinvaidya17) [reply src/anomalib/trainer/connectors/thresholding_connector.py:81]: Yes

**Review by samet-akcay — APPROVED** [review]
lgtm, thanks a lot!

**Review by djdameln — APPROVED** [review]
Thanks!

**Pull request merged**